### PR TITLE
Mirror `./examples/community` folder on HF

### PR DESCRIPTION
**Pull request by Wauplin**
Related to [internal slack thread](https://huggingface.slack.com/archives/C03HBN1C8CW/p1717672763942379).

**HF mirror example:** https://huggingface.co/datasets/Wauplin/diffusers-community-pipelines-mirror. Each tag/branch is a subfolder with all the pipelines in it.

This PR adds a new CI workflow to host the community pipelines folder (`./examples/community`) as an HF dataset. The HF dataset will only be a mirror of the content on Github and should be automatically updated by this CI. Action is triggered on commits on new tags (i.e. on new releases) and on commits on `main` updating the `./examples/community` folder.

Once this will be done, we will be able to download community pipeline modules directly from the HF Hub instead of relying on Github files. This should simplify the logic as we will be able to use `hf_hub_download` which handles `cache_dir`, `force_download`, `local_files_only`, etc... natively. Downloading from Github currently requires the deprecated [`cached_download`](https://github.com/huggingface/diffusers/blob/a3faf3f260101c2bc09112ebb514d1e0d0220e1a/src/diffusers/utils/dynamic_modules_utils.py#L287)  that will soon be removed from `huggingface_hub`.

---

**TODO** **_before merging_:** (by a `diffusers` maintainer)
- [x] Create a dataset repo in `diffusers/community-pipelines-mirror` in the [diffusers organization](https://huggingface.co/diffusers) on the Hub.
- [x] Create a fine-grained token for https://huggingface.co/diffusers-bot that has write access only to this dataset repo
- [x] Set this newly created token as `HF_TOKEN_MIRROR_COMMUNITY_PIPELINES` in Github secret

**TODO _once merged_:**
- [ ] upload all previous tags to the dataset mirror
- [ ] get rid of `cached_download` in a follow-up PR. See draft: https://github.com/huggingface/diffusers/pull/8419

### CODE REVIEW

File: .github/workflows/mirror_community_pipeline.yml
Comment (sayakpaul) [inline .github/workflows/mirror_community_pipeline.yml:41]: Would be nice provide an example for `PATH_IN_REPO` in the comment for the different conditions we're adding here.

File: .github/workflows/mirror_community_pipeline.yml
Comment (sayakpaul) [inline .github/workflows/mirror_community_pipeline.yml:51]: Destination dataset repo id to be updated.

**Review by sayakpaul — APPROVED** [review]
LGTM, thanks for this!

I have taken care of the todos (before merging one), too.

File: .github/workflows/mirror_community_pipeline.yml
Comment (Wauplin) [reply .github/workflows/mirror_community_pipeline.yml:42]: Good idea. Added an example in https://github.com/huggingface/diffusers/pull/8417/commits/cb0bf321aceecd6e1c9d5328dfca4b30998494d0

File: .github/workflows/mirror_community_pipeline.yml
Comment (Wauplin) [reply .github/workflows/mirror_community_pipeline.yml]: Good catch! 

Added it in https://github.com/huggingface/diffusers/pull/8417/commits/2d9a9ecfffc73236dc8c067f3d4635ac2b0148ca

**Comment by Wauplin** [comment]
Thanks for the review and already handling the TODOs @sayakpaul! :heart:
I've addressed your comments so we're good to merge I think :)

**Comment by sayakpaul** [comment]
No worries. Will merge once the CI is green (although it should not matter since the CI is not supposed to be triggered during PRs). 

From the todos after merging:

> upload all previous tags to the dataset mirror

What exactly needs to be done here?

**Comment by Wauplin** [comment]
> What exactly needs to be done here?

I actually just pushed a new commit (https://github.com/huggingface/diffusers/pull/8417/commits/72783bb1d7d0703be2c2039d1151678dbb7e4d71) that will allow us to trigger this workflow manually. What we need to do is trigger a deploy for every existing tag + for the main branch to populate the dataset repo. The CI will then take care of updating it when needed. I can take care of that part I think and will let you know once it's done.

**Comment by Wauplin** [comment]
(sorry, just realized the yml is incorrect. Converted to draft to avoid it being merged)

**Comment by Wauplin** [comment]
Ok, sorry for the mess. I created and merged https://github.com/huggingface/diffusers/pull/8425, https://github.com/huggingface/diffusers/pull/8426 and https://github.com/huggingface/diffusers/pull/8427 to fix a few things. Files are getting correctly hosted on the Hub: https://huggingface.co/datasets/diffusers/community-pipelines-mirror/tree/main :tada: 

Workflow can be manually triggered on https://github.com/huggingface/diffusers/actions/workflows/mirror_community_pipeline.yml.
Next time you do a new release, would be good to check that everything is correctly uploaded as well.

**Comment by sayakpaul** [comment]
Cool, would you mind updating our Slack channel about this? This is very nice move, indeed. 

> Next time you do a new release, would be good to check that everything is correctly uploaded as well.

Perhaps we could set up automated reporting for this?

**Comment by Wauplin** [comment]
> would you mind updating our Slack channel about this? 

Will do!

Perhaps we could set up automated reporting for this?

A CI checking if the CI failed?

**Comment by sayakpaul** [comment]
> A CI checking if the CI failed?

If the CI isn't successful, we will report the error to a Slack channel.

**Comment by Wauplin** [comment]
Not a priority indeed! Good luck for the upcoming releases

**Pull request merged**
